src/teleop/gui/src/components/ArmControls.vue

```diff
@@ -176,17 +179,22 @@ export default {
 
   methods: {
     updateArmMode: function () {
+      console.log("Updating arm mode");
       const armData = {
-        data: this.arm_controls,
+        mode: this.arm_controls
       };
-      var armcontrolsmsg = new ROSLIB.Message(armData);
-      this.armcontrols_pub.publish(armcontrolsmsg);
+      var armcontrolsmsg = new ROSLIB.ServiceRequest(armData);
+      this.ra_mode_service.callService(armcontrolsmsg, (response) => {
+        if (!response.success) {
+          alert("Failed to change arm mode");
```

Comment:
This is not ideal since it doesn't update the value upon failure, though I guess the user is still alerted so it's not the end of the world. Probably fine as is